packages/block-editor/src/components/link-control/index.js

```diff
@@ -118,9 +118,10 @@ function LinkControl( {
 			'is-loading': isLoading,
 		} );
 
+		const searchText = inputValue.trim();
```

Comment:
Let's move this logic into `URLInput`. Please see https://github.com/WordPress/gutenberg/pull/19832#discussion_r378782848